### PR TITLE
doc: Update IVSHMEM tutorial

### PR DESCRIPTION
**Pull request by amyreye**
- Update overview, dependencies and constraints
- Update to match Configurator UI instead of manually editing XML files
- Remove architectural details and instead point to high-level design documentation

Signed-off-by: Reyes, Amy <amy.reyes@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

File: doc/tutorials/enable_ivshmem.rst [hunk 1]
```diff
@@ -1,229 +1,200 @@
...
-   - For VM1 use ``ls -lh /sys/bus/pci/devices/0000:00:06.0/uio``
-   - For VM2 use ``ls -lh /sys/bus/pci/devices/0000:00:05.0/uio``
+   a. As IVSHMEM is a PCI device that shares memory inside a virtual machine,
+      the application running on the VM can access it as a standard PCI device.
+      The application needs to design a data transfer notification mechanism
+      between the VMs. See :ref:`inter-vm_communication_ivshmem_app`. In this example, the application is ``ivshmem_test.bin``.
 
-HV-Land Example
-===============
+   #. Check the inter-VM shared memory vail. Run the following commands in
```
Comment (dbkinder) [inline doc/tutorials/enable_ivshmem.rst:134]: vail?

File: doc/tutorials/enable_ivshmem.rst [hunk 2]
```diff
@@ -1,229 +1,200 @@
...
-             <IVSHMEM_ENABLED>y</IVSHMEM_ENABLED>
-             <IVSHMEM_REGION>hv:/shm_region_0, 2, 0:2</IVSHMEM_REGION>
-      </IVSHMEM>
+   #. Run the following command in POST_STD_VM2 to send the file:
 
-2. Build ACRN based on the XML configuration for hybrid_rt scenario on whl-ipc-i5 board::
+      .. code-block:: console
 
-      make BOARD=whl-ipc-i5 SCENARIO=<path/to/edited/scenario.xml>
+         root@acrn-Standard-PC-i440FX-PIIX-1996:~# ./ivshmem_test.bin uio0 tx test.file
```
Comment (dbkinder) [inline doc/tutorials/enable_ivshmem.rst:151]: this "ivhsmem_test.bin" executable doesn't exist and there are no instructions how to create it.

File: doc/tutorials/enable_ivshmem.rst [hunk 3]
```diff
@@ -1,229 +1,200 @@
...
-   The ``X`` in ``uioX`` above, is a number that can be retrieved using the
-   ``ls`` command:
+#. To verify inter-VM shared memory communication:
 
-   - For VM1 use ``ls -lh /sys/bus/pci/devices/0000:00:06.0/uio``
-   - For VM2 use ``ls -lh /sys/bus/pci/devices/0000:00:05.0/uio``
+   a. As IVSHMEM is a PCI device that shares memory inside a virtual machine,
+      the application running on the VM can access it as a standard PCI device.
+      The application needs to design a data transfer notification mechanism
+      between the VMs. See :ref:`inter-vm_communication_ivshmem_app`. In this example, the application is ``ivshmem_test.bin``.
```
Comment (dbkinder) [inline doc/tutorials/enable_ivshmem.rst:132]: There are no instructions in that doc for how to create this app

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by dbkinder — APPROVED** [review]
OK, thanks.

**Pull request merged**